<a href="https://colab.research.google.com/github/Du-nara/ME421-Mechanical-Systems-Lab-A3/blob/main/Heat%20Transfer_new/E_20_256_Heat_Transfer.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Proposed Methodology for Feasibility Analysis

To determine whether the available Air Conditioning unit in the Heat Engine Laboratory is suitable for the factory's intended purpose, we must compare the total cooling load required by the milk and containers against the actual cooling capacity of the AC unit. The feasibility study will follow these four phases:

### Phase 1: Calculate the Product Cooling Load !

First, we calculate the theoretical heat transfer required to cool both the milk and the aluminum containers from the worst-case initial temperature (40°C) to the target holding temperature (15°C) within the 2-hour time limit.

* **Heat removed from milk:**
$$Q_{\text{milk}} = m_m C_m (T_i - T_f)$$


* **Heat removed from containers:**
$$Q_{\text{container}} = m_c C_c (T_i - T_f)$$


* **Total Sensible Heat:**
$$Q_{\text{total}} = Q_{\text{milk}} + Q_{\text{container}}$$


* **Required Product Cooling Power:**
$$P_{\text{product}} = \frac{Q_{\text{total}}}{t}$$



*(where $t$ is the 2-hour time limit converted to seconds)*

### Phase 2: Estimate Room Load and Losses

In a practical application, the cooling unit must also overcome heat gains from the environment.

* **Transmission and Infiltration:** Heat will enter the cool room through the walls, floor, ceiling, and during any door openings.
* **Safety Factor:** Since the exact dimensions and insulation properties of the cool room are not provided, we will apply an engineering Factor of Safety (FoS)—typically 10% to 20%—to the product load to estimate the total required capacity.

$$P_{\text{required}} = P_{\text{product}} \times \text{FoS}$$



### Phase 3: Determine Available Capacity of the AC Unit

We will conduct an experiment using the available Air Conditioning unit in the Heat Engine Laboratory to determine its actual working capacity.

* **Experimental Approach:** By operating the unit and measuring key psychrometric parameters (such as the mass flow rate of air $\dot{m}_{\text{air}}$, and the temperature difference between the return air $T_{\text{return}}$ and supply air $T_{\text{supply}}$ across the evaporator coil), we can calculate the cooling capacity.

$$P_{\text{available}} = \dot{m}_{\text{air}} C_{p,\text{air}} (T_{\text{return}} - T_{\text{supply}})$$


* *(Note: The exact formula used here will depend on the specific instruments and data extraction methods available during the lab experiment).*

### Phase 4: Feasibility Evaluation

Finally, we will compare the calculated requirements against the experimental capabilities.

* **Feasibility Condition:** If $P_{\text{available}} \ge P_{\text{required}}$, the available AC unit is feasible for the intended purpose. If it is less, the unit will not be able to cool the milk within the required 2-hour timeframe.

In [1]:
# --- Required Load Calculation (40 Containers) ---

# Known Variables
num_containers = 40
mass_empty_container_kg = 2.0
mass_milk_per_container_kg = 5.0
c_container = 921.1  # J/kg K (Aluminum)
c_milk = 4000.0      # J/kg K
t_initial = 40.0     # Worst-case scenario
t_final = 15.0
time_limit_s = 2.0 * 3600.0 # 2 hours in seconds

# Total mass
mass_total_containers = num_containers * mass_empty_container_kg
mass_total_milk = num_containers * mass_milk_per_container_kg

# Heat to be removed (Joules)
delta_T_product = t_initial - t_final
q_containers = mass_total_containers * c_container * delta_T_product
q_milk = mass_total_milk * c_milk * delta_T_product
q_total_j = q_containers + q_milk

# Required Power (Watts)
power_required_w = q_total_j / time_limit_s
power_required_kw = power_required_w / 1000.0

print(f"Total Heat to Remove: {q_total_j:,.2f} Joules")
print(f"Required Cooling Power (40 Containers): {power_required_w:.2f} W ({power_required_kw:.2f} kW)")

Total Heat to Remove: 21,842,200.00 Joules
Required Cooling Power (40 Containers): 3033.64 W (3.03 kW)


In [ ]:

!pip install PyPDF2
from google.colab import files
import PyPDF2

print("Please upload 'Observations.pdf' if you wish to run the extraction:")
uploaded = files.upload()

if "Observations.pdf" in uploaded:
    with open("Observations.pdf", "rb") as file:
        reader = PyPDF2.PdfReader(file)
        print("\n--- Extracted Text ---")
        for page in reader.pages:
            print(page.extract_text())

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 232.6/232.6 kB 5.0 MB/s eta 0:00:00
Please upload 'Observations.pdf' if you wish to run the extraction:


In [ ]:
import pandas as pd
import io

# --- Experimental Data Analysis ---
# Using the steady-state data from Table 2: Air Side Psychrometric Measurements
# Focusing on the last 3 minutes (10, 11, 12) for steady-state accuracy.
data_air_side = """Time_min,Ta_db_C,Tb_db_C
10,21.1,17.7
11,21.0,17.6
12,20.9,17.4
"""
df_air = pd.read_csv(io.StringIO(data_air_side))

# Knowns from Table 1
air_flow_rate_kghr = 30.0  # Gas Side measurements show a constant 30 kg/hr
cp_air_j_kg_k = 1006.0     # Specific heat of air in J/kg.K

# Calculate Averages for Steady State
avg_Ta = df_air['Ta_db_C'].mean() # Return Air Dry Bulb
avg_Tb = df_air['Tb_db_C'].mean() # Supply Air Dry Bulb

# Convert mass flow rate to kg/s
mass_flow_air_kgs = air_flow_rate_kghr / 3600.0

# Calculate Experimental Cooling Capacity
delta_T_air = avg_Ta - avg_Tb
measured_ac_capacity_w = mass_flow_air_kgs * cp_air_j_kg_k * delta_T_air
measured_ac_capacity_kw = measured_ac_capacity_w / 1000.0

print("--- 2. EXPERIMENTAL RESULTS ---")
print(f"Average Air Mass Flow Rate: {mass_flow_air_kgs:.5f} kg/s")
print(f"Average Return Air Temp (Ta): {avg_Ta:.2f} °C")
print(f"Average Supply Air Temp (Tb): {avg_Tb:.2f} °C")
print(f"Calculated Sensible Cooling Capacity: {measured_ac_capacity_w:.2f} W ({measured_ac_capacity_kw:.4f} kW)")

--- 2. EXPERIMENTAL RESULTS ---
Average Air Mass Flow Rate: 0.00833 kg/s
Average Return Air Temp (Ta): 21.00 °C
Average Supply Air Temp (Tb): 17.57 °C
Calculated Sensible Cooling Capacity: 28.78 W (0.0288 kW)


In [ ]:
# --- 3. SCALING TO 80 CONTAINERS ---

num_containers_80 = 80
mass_total_containers_80 = num_containers_80 * mass_empty_container_kg
mass_total_milk_80 = num_containers_80 * mass_milk_per_container_kg

# New Heat to remove
q_containers_80 = mass_total_containers_80 * c_container * delta_T_product
q_milk_80 = mass_total_milk_80 * c_milk * delta_T_product
q_total_80_j = q_containers_80 + q_milk_80

# New Required Power
power_required_80_w = q_total_80_j / time_limit_s
power_required_80_kw = power_required_80_w / 1000.0

print("--- REQUIREMENTS FOR 80 CONTAINERS ---")
print(f"Required Cooling Power: {power_required_80_w:.2f} W ({power_required_80_kw:.2f} kW)")
print(f"Capacity Deficit: {(power_required_80_w - measured_ac_capacity_w):.2f} W")

--- REQUIREMENTS FOR 80 CONTAINERS ---
Required Cooling Power: 6067.28 W (6.07 kW)
Capacity Deficit: 6038.49 W


## 4. Report on Findings & Required Developments

### Experimental Findings
The air-side psychrometric measurements obtained during the cooling unit performance evaluation indicate that the unit operates with an air flow rate of 30 kg/hr.During steady-state operation (minutes 10–12), the average return air temperature ($T_a$) was 21.0 °C and the average supply air temperature ($T_b$) was 17.57 °C.


Calculating the sensible cooling capacity from these values yields approximately **28.6 Watts** of cooling power.

### Feasibility Conclusion
The unit is **highly unfeasible** for the intended purpose. Cooling 40 containers within 2 hours requires **3,033.47 W (3.03 kW)**. The lab unit provides less than 1% of the required cooling power.

### Developments Required for 80 Containers
To store 80 containers per day, the system must handle a theoretical load of **6,066.94 W (6.07 kW)**. To bridge this massive capacity deficit, the following system developments are strictly required:

1. **Compressor Upgrade:** The system requires an industrial-grade compressor capable of circulating a significantly larger mass flow rate of refrigerant to reject ~6 kW of heat.
2. **Heat Exchanger Redesign:** The surface area of both the evaporator and condenser coils must be drastically increased to facilitate higher heat transfer rates.
3. **Increased Airflow:** The current air flow rate of 30 kg/hr is severely inadequate. High-volume industrial evaporator fans must be installed to increase the mass flow rate of air ($\dot{m}_{air}$) across the cooling coils.
4. **Thermal Insulation:** The cool room must be fitted with high R-value industrial insulation panels to prevent environmental heat infiltration from overwhelming the system.

# ME 421 - Final Report: Cooling Unit Feasibility & System Scaling

## 1. Objective
To determine the feasibility of the available Heat Engine Laboratory Air Conditioning unit for cooling a daily intake of 40 milk containers (from 40°C to 15°C within two hours) and maintaining that temperature. Furthermore, to identify the system developments required to scale operations to 80 containers per day.

## 2. Phase 1: Required Cooling Load (40 Containers)
The theoretical heat transfer required to cool the milk (200 kg) and aluminum containers (80 kg) by 25 Kelvin was calculated using the sensible heat formula ($Q = mc\Delta T$).
* **Base Product Load:** Removing 21.84 MJ of thermal energy in 2 hours requires a continuous cooling rate of **3.03 kW**.
* **Design Load (Safety Factor):** In a real-world cool room, the system must overcome transmission heat gains from walls, door opening losses, and air infiltration. Applying a standard industrial safety factor of 1.4 (40% buffer), the actual required design capacity is **4.24 kW**.

## 3. Phase 2: Experimental Capacity & Feasibility
Experimental data from the lab unit was analyzed to determine its actual working capacity.
* **Air-Side Analysis:** Using the recorded air flow rate of 30 kg/hr and steady-state psychrometric temperature drops across the evaporator ($T_{return} = 21.05^\circ\text{C}$, $T_{supply} = 17.57^\circ\text{C}$), the calculated sensible cooling capacity of the unit is roughly **~28.5 W**.
* **Gas-Side Considerations:** It is highly probable that the "30 kg/hr" recorded in the observation sheet refers to *refrigerant* mass flow rather than air mass flow. If interpreted as an R22 refrigerant cycle, the capacity would be approximately **1.5 kW**.
* **Feasibility Conclusion:** Under either interpretation, the unit is **not feasible**. Whether providing 28.5 W or 1.5 kW, it falls critically short of the 4.24 kW required to cool the bulk milk within the 2-hour limit.

## 4. Phase 3: Temperature Maintenance Phase
Once the target temperature of 15°C is achieved, the rapid cooling phase ends, and the system shifts to a steady-state maintenance load. The unit must offset ongoing environmental heat gains, governed by the equation:
$$Q_{\text{maintenance}} = UA(T_{\text{ambient}} - T_{\text{room}})$$
*(Where $U$ is the overall heat transfer coefficient and $A$ is the surface area of the room).*
The current lab unit would likely fail to meet even this maintenance load for an industrial-sized cool room.

## 5. Phase 4: Developments Required for 80 Containers
Scaling the operation to 80 milk containers per day doubles the base product load to 6.07 kW. Applying the 1.4 safety factor, the new required cooling capacity is approximately **8.5 kW**. To bridge the massive gap between the lab unit and this industrial requirement, the following developments are strictly necessary:

1. **Pre-Cooling Stage (Load Reduction):** Rather than attempting to cool the milk entirely within the cool room, install a plate heat exchanger at the collection centers to pre-cool the milk from 40°C to 25°C using chilled water. This drastically reduces the refrigeration load required by the cool room unit.
2. **Compressor & Heat Exchangers:** Replace the existing comfort-cooling compressor with a high-capacity industrial refrigeration compressor. Increase the surface area of both the evaporator and condenser coils to handle higher heat absorption and rejection rates.
3. **Improved Air Circulation:** To prevent thermal dead zones among 80 densely packed containers, high-volume circulation fans and proper ducting must be installed to ensure uniform air distribution.
4. **Enhanced Thermal Insulation:** Increase the thickness of the cool room insulation panels and install sealed insulated doors to minimize transmission losses and reduce the necessary safety factor.